packages/@uppy/core/src/index.js

```diff
@@ -100,8 +99,9 @@ class Uppy {
     }
 
     // Merge default options with the ones set by user
-    this.opts = Object.assign({}, defaultOptions, opts)
-    this.opts.restrictions = Object.assign({}, defaultOptions.restrictions, this.opts.restrictions)
+    this.opts = {}
+    this.setOptions(defaultOptions)
+    this.setOptions(opts)
```

Comment:
I think we can safely do this here:
```js
this.opts = defaultOptions
this.setOptions(opts)
```
since `defaultOptions` is a local object that is created inside the constructor already.
Then `setOptions()` will only do the locale init once. Right now we're creating a Translator instance and then immediately throwing it away and creating a second one.